orttraining/orttraining/training_api/include/onnxruntime_training_c_api.h

```diff
@@ -221,6 +221,54 @@ struct OrtTrainingApi {
   */
   ORT_API2_STATUS(SchedulerStep, _Inout_ OrtTrainingSession* sess);
 
+  /** \brief Retrieves the size of all the parameters.
+  *
+  * Calculates the size of all the parameters for the training session.
+  * When 'trainable_only' is true, the size is calculated for trainable params only.
+  *
+  * \param[in] sess The training session.
+  * \param[in] trainable_only Whether to skip non-trainable parameters
+  *
+  * \snippet{doc} snippets.dox OrtStatus Return Value
+  *
+  */
+  ORT_API2_STATUS(GetParametersSize, _Inout_ OrtTrainingSession* sess,
+                  _Out_ size_t* out, bool trainable_only);
```

Comment:
This option is in order to maintain some flexibility in the APIs that we provide. Since FL is still an upcoming field, there might be such a need in the future where the entire model on the client needs to be updated(including frozen parameters) by the server.